SmilesMiles/View/TripDetail.swift

```diff
@@ -37,14 +39,17 @@ struct TripDetail: View {
         }
         .accentColor(.green)
         .navigationBarTitle(tripName)
-        .navigationBarItems(trailing: Button(action: deleteTrip) {
+        .navigationBarItems(trailing: Button(action: {
+            let viewModel = TripDetailViewModel(trips: $trips, numberOfTrips: $numberOfTrips, tripName: tripName)
+            viewModel.deleteTrip()
```

Comment:
Can you please check if you can pass the trip to be deleted directly to deletTrip() instead of passing it to the TripDetailViewModel()?

Reply:
Resolved. However, there is a warning that publishing changes from within view updates is not allowed, this will cause undefined behavior.